db/contentGraph.ts

```diff
@@ -46,58 +53,175 @@ export const getGrapherSlugs = (content: string | null): Set<string> => {
     return slugs
 }
 
-export const getContentGraph = once(async () => {
-    const orderBy = "orderby:{field:MODIFIED, order:DESC}"
-    const entries = await getDocumentsInfo(
-        WP_PostType.Page,
-        "",
-        `categoryId: ${ENTRIES_CATEGORY_ID}, ${orderBy}`
-    )
-    const posts = await getDocumentsInfo(WP_PostType.Post, "", orderBy)
-    const documents = [...entries, ...posts]
+const throwAllButConflictError = (err: unknown): void => {
+    if (!(err instanceof ConflictError)) {
+        throw err
+    }
+}
+
+export const addDocumentsToGraph = async (
+    documents: DocumentNode[],
+    graph: any
+): Promise<void> => {
     for (const document of documents) {
+        // Create the parent topics first (add records with the only available
+        // referential information - id - then update the records when going
+        // through the parent topic as a document)
+        if (document.parentTopics) {
+            for (const parentTopic of document.parentTopics) {
+                try {
+                    await graph.create(GraphType.Document, {
+                        id: parentTopic,
+                    })
+                } catch (err) {
+                    // If the document has already being added as a parent topic of
```

Comment:
I don't entire understand this - could you try to phrase this more explicitly? What I am unsure about is: is it a problem if the current element is also a parent topic? I.e. can there only be one level of parents and children? 

And relatively regardless of this I wonder - are we throwing away information here if there is a conflict (i.e. in effect a parent relationship found earlier is preferred over others) or is this simply to exclude cycles?

Reply:
I think it's just to avoid duplicates – this loop iterates through every node and ensures the parents are also added as nodes (not concerned with adding _edges_ yet here). The parents could've been added previously by another node. (Also just the `id` is added, in order to make it possible to detect duplicates – same `id` leads to `ConflictError` thrown.)

The relationships are created implicitly by `fortune` when a node is created with a relationship field populated (`parentTopics` in this case).

@mlbrgl let me know if that's not quite right!

Reply:
Thanks Daniel! The location of this comment got shifted - I had remarked this on this a few lines above where the comment explains what happens here :). What Daniel outlined makes sense but maybe the comment could guide the reader a bit more along those lines?

Reply:
@danielgavrilov correct!

@danyx23 thanks for the suggestion, I've added a longer comment. More specifically, to your questions:
> is it a problem if the current element is also a parent topic? I.e. can there only be one level of parents and children?

No that's not a problem, I'd even say it is expected. Theoritically, all nodes could be chained together, there is no conceptual limit.

> And relatively regardless of this I wonder - are we throwing away information here if there is a conflict (i.e. in effect a parent relationship found earlier is preferred over others) or is this simply to exclude cycles?

All relationships as well as node attributes are recorded. Nothing is discarded.